smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/traits/EndpointModifierIndex.java

```diff
@@ -22,15 +25,22 @@
  * Endpoint modifier traits are traits that are marked by {@link EndpointModifierTrait}
  */
 public final class EndpointModifierIndex implements KnowledgeIndex {
+    private static final Logger LOGGER = Logger.getLogger(EndpointModifierIndex.class.getName());
 
     private final Map<ShapeId, Map<ShapeId, Trait>> endpointModifierTraits = new HashMap<>();
 
     public EndpointModifierIndex(Model model) {
         for (ServiceShape serviceShape : model.getServiceShapes()) {
             Map<ShapeId, Trait> result = new TreeMap<>();
             for (Trait trait : serviceShape.getAllTraits().values()) {
-                Shape traitShape = model.expectShape(trait.toShapeId());
-                if (traitShape.hasTrait(EndpointModifierTrait.ID)) {
+                Optional<Shape> traitShape = model.getShape(trait.toShapeId());
+                if (!traitShape.isPresent()) {
```

Comment:
Prefer using `Optional::isEmpty()`:

```suggestion
                if (traitShape.isEmpty()) {
```

Reply:
Apparently this is only available in Java 11.